resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+
+- System integrators : System integrators are partners who help enterprises integrate proprietary services with cloud native applications. They are responsible for designing and building cloud native applications on behalf of their customers.
+
+- Developers : Developers are responsible for building the cloud native applications. They are responsible for writing the code, designing and maintaining the applications.
+
+- Open-source contributors : Cloud native open-source contributors can be any of the above personas who are interested in contributing to the cloud native projects. 
+
+### Positive user outcomes
+
+- Author and deploy : I can bring my own applications with custom resources/services and integrate with Radius seamlessly. I don’t have to write Go code to integrate with Radius components and can use simple spec to generate the resource definition and integrate with Radius 
+
+- Recipes for user defined types: I can create Recipes for the custom resource types and deploy via Radius 
+
+- OSS contributions: I can create and contribute Recipes for the custom resource types for the community to use.
+
+## Key scenarios
+
+### Scenario 1: Deb integrates the Budgets app with Radius
+
+Deb, a platform engineer at a Financial Services company, wants to integrate the Budgets app with Radius. The Budgets app relies on an internal messaging service called Plaid. Deb wants to use Plaid as a resource type in Radius to deploy the application seamlessly. He needs a way to define and use Plaid as a custom resource type in Radius without having to write Go code. This will enable him to leverage all the features of Radius, such as Recipes, Connections, and the rad CLI, with ease and flexibility.
+
+### Scenario 2: Amy contributes and open-sources PostgreSQL support to Radius
+
+Amy is a system integrator who helps customers build cloud native applications on AWS. Amy heard about Radius and starts integrating their customer workloads in Radius. She sees a lot of her scenarios involve PostgreSQL and finds Radius doesn’t support that yet. She wants to contribute the PostgreSQL support to Radius and open source it so anyone can use it
+
+### Scenario 3: Raj publishes Recipes for proprietary services
+
+Raj is a partner or a system integrator who helps enterprises integrate proprietary service for eg : Oracle database in their cloud native applications. Raj tries out Radius and wants to write Recipes to create an Oracle Database with a standard set of best practices and policies integrated. He works with a lot of customers who are wanting to use the Oracle Database recipe in their applications.
```

Comment:
> He works with a lot of customers who are wanting to use the Oracle Database recipe in their applications.
 
I think this scenario might be even more illustrative if Raj isn't just an integrator, but perhaps someone who works on the Oracle Database team and is looking to expand their adopter/user base. By contributing an Oracle Database recipe, they are hoping to lower the barrier for entry and attract Radius users to using Oracle products. This would give this scenario a "selfishly contributes to Radius" story where open-source and enterprise collaborate for a win-win situation.

Reply:
flush some more details aeround soultion provider on the benefits